### DISCUSSION TITLE
KY 040 not changing Volume

### DISCUSSION THREAD

**Discussion by Guddi11** [MiczFlor/RPi-Jukebox-RFID#1674] [Q&A]
Hi,
I am trying to use the often used setup with the KY040 as Volume control and the Hifiberry Miniamp as amplifier. Unfortunately I cannot get the knob to make the sound go up/down. 
The KY040 itself seems to be working - I put a logging handler temporarily to the rotary_encoder.py, that streams to stdout and I get the expected notifications. 

Is there any config, that I still need to set and overlooked?
I'd really appreciate some hint. Thanks!

I am using an RPi Zero W, Headless Raspberry-Pi OS, RPi-Jukebox Image (v2) from about 4 weeks ago.
This is my gpio_settings.ini
```ini
[DEFAULT]
enabled : True

[VolumeControl]
enabled : True
Type : RotaryEncoder
Pin1 : 5
Pin2 : 6
timeBase: 0.2
functioncall1 : functionCallVolU
functioncall2 : functionCallVolD

[PlayPause]
enabled : True
Type : Button
pin : 27
hold_time : 0.3
functioncall : functionCallPlayerPause
pull_up_down : pull_up

[NextSong]
enabled : True
type : Button
pin : 22
hold_time : 0.3
functioncall : functionCallPlayerNext

[PrevSong]
enabled : True
type : Button
pin : 23
hold_time : 0.3
functioncall : functionCallPlayerPrev
```

**Comment by gsandreas** [comment]
Hi, 

Got an similar problem. 

Try 
pi@Raspi:~ $ raspi-gpio get 

and check, if all of our used GPIO are free as an input. 

Greetz
Andreas

**Reply by Guddi11** [reply]
Thanks a lot!
Due to some time constraints (Xmas…) I switched to buttons now and also modified my custom pcb. That’s why I can’t check that with the current setup. The buttons work flawlessly anyway so probably it’s not a pin assignment thing. Maybe I buy components for a test setup ;)

Merry Xmas.

**Comment by Lippsson** [comment]
My settings works fine:

```
[VolumeControl]
enabled: True
Type: RotaryEncoder
PinUp: 5
PinDown: 6
pull_up_down: pull_up
timeBase: 0.2
functionCallDown: functionCallVolD
functionCallUp: functionCallVolU
```
